src/NServiceBus.Core/Persistence/PersistenceExtentions.cs

```diff
@@ -19,10 +19,21 @@ public class PersistenceExtentions<T, S> : PersistenceExtentions<T>
         /// <summary>
         /// </summary>
         /// <param name="settings"></param>
-        public PersistenceExtentions(SettingsHolder settings)
-            : base(settings)
+        public PersistenceExtentions(SettingsHolder settings) : base(settings, typeof(S))
         {
         }
+
+        /// <summary>
+        /// Defines the list of specific storage needs this persistence should provide
+        /// </summary>
+        /// <param name="specificStorages"></param>
+        /// <returns></returns>
+        /// <exception cref="InvalidOperationException"></exception>
+        [ObsoleteEx(RemoveInVersion = "7.0", TreatAsErrorFromVersion = "5.2")]
+        public new PersistenceExtentions<T> For(params Storage[] specificStorages)
+        {
+            throw new InvalidOperationException("Do not invoke .For() when StorageType is already specified.");
```

Comment:
@SeanFeldman if we are going to release this feature as a minor release we need to make this method backwards compatible.

I understand the user could do something like:
```c#
config.UsePersistence<AzureStoragePersistence, StorageType.Subscriptions>().For(Storage.Sagas);
```
But this is a scenario that is not possible, users wouldn't be able to recompile using the `For` and if they just drop in the minor dll, the code would still be using the old syntax, so, I think it is safe to re-add the old code here so that this method still works, thoughts?